FHIR-SW-Development-Guide.ipynb

```diff
@@ -0,0 +1,1560 @@
+    "    - Tests to validate FHIR model deployment\n",
+    "\n",
+    "\n",
+    "4. [Tutorial Data](data) (`data`)\n",
+    "    - The tutorial's completed FHIR model resources.\n",
+    "    \n",
+    "⚡️ **The NCPI FHIR model must follow a specific file and folder hierarchy because model validation is done using the standard HL7 validator called the ImplementationGuide publisher. This toolexpects a particular structure.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "### Create Feature Branch\n",
+    "\n",
+    "Create a feature branch for the work you will do in this tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true
+   },
+   "source": [
+    "## FHIR Concept Refresher"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "fatal: A branch named 'add-patient-species' already exists.\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Ensure you are in the ncpi-model-forge folder, then make a feature branch\n",
+    "!git checkout -b add-patient-species"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "Before we begin, let's do a quick refresher on some key modeling concepts in FHIR: \n",
+    "\n",
+    "### FHIR Base Model\n",
+    "- FHIR comes with a model consisting of conformance resources and terminology resources for the core FHIR   resources (e.g. Patient, Specimen, Observation, etc.)\n",
+    "\n",
+    "### Conformance Resources\n",
+    "- An entity in FHIR is called a `Resource`. Examples: `Patient`, `Specimen`, `StructureDefinition`\n",
+    "- The phrase \"FHIR Model\" refers to a set of FHIR conformance resources\n",
+    "- Types of conformance resources include `StructureDefinition`, `CodeSystem`, `SearchParameter`.\n",
+    "- A `StructureDefinition` is essentially a schema. \n",
+    "- A `StructureDefinition` can represent the schema for a particular FHIR resource (e.g. Patient) or an attribute  on a FHIR resource (e.g. biological-sex)\n",
+    "- A `StructureDefinition` that represents an attribute on a FHIR resource (e.g. biological-sex) is called an Extension\n",
+    "- An Extension based `StructureDefinition` is used to add a new attribute to a FHIR resource\n",
+    "- A non-Extension based `StructureDefinition` is often called a `profile`\n",
+    "\n",
+    "### Terminology Resources\n",
+    "- A `CodeSystem` is a FHIR resource that represents an ontology\n",
+    "- A `CodeSystem` contains concepts and their codes \n",
+    "- A `ValueSet` is a restricted set of codes (e.g. `male` | `female`) from one or more `CodeSystems`\n",
+    "- A `ValueSet` can be bound to an attribute on a FHIR resource to restrict the values of the attribute\n",
```

Comment:
We may want to mention that a `CodeSystem` can also be bound to a `ValueSet` so that actual codes doesn't necessarily have to be specified in a `ValueSet`?

Reply:
I think that's a detail that should go into the FHIR 101 notebook. These bullets are just meant to be the basics